pinctrl/pinctrl_utils.go

```diff
@@ -53,7 +78,7 @@ func cleanFilePath(nodePath string) string {
 }
 
 // We look in the 'aliases' node at the base of proc/device-tree to determine the full file path required to access our GPIO Chip.
-func (b *pinctrlpi5) findPathFromAlias(nodeName string) (string, error) {
+func findPathFromAlias(nodeName, dtBaseNodePath string) (string, error) {
 	dtNodePath := dtBaseNodePath + "/aliases/" + nodeName
```

Comment:
no, the `dtBaseNodePath` will be the same for all boards, but not all device trees have aliases and the `nodeName`  is hardcoded to be`gpio0`, which could be different on other boards.

Reply:
oh `nodeName` shouldn't be hardcoded here, the implementer sets it as part of their config when trying to use pinctrl

Reply:
oh okay I see, still not every dt has aliases so maybe we should avoid using it. Honestly making the device tree parsing general enough for all boards might be very difficult since every device tree ive looked at have very different formats.

Reply:
For example, the orange pi has an `alias` node, but it does not include aliases for the gpio chips so this line would not work.

Reply:
you can just have the user supply the whole path to the gpiochip in the config instead of getting it from the alias

Reply:
I'll wait to look at this particular file to give you time to change this.

Reply:
so I didn't want to delete the alias method outright because it does look clean in the config, so instead I added a bool to signal whether the gpio chip path should assume an alias. 

Messing with the code more I am alittle bit concerned that the ranges/address-cells won't be the same for other boards, but I think we can worry about that when we go to implement other boards? also happy to try making those fields optional